parquet/src/main/java/org/apache/iceberg/data/parquet/BaseParquetReaders.java

```diff
@@ -359,10 +250,10 @@ public ParquetValueReader<?> primitive(
 
       ColumnDescriptor desc = type.getColumnDescription(currentPath());
 
-      if (primitive.getOriginalType() != null) {
+      if (primitive.getLogicalTypeAnnotation() != null) {
```

Comment:
I agree with this change, but please point these kinds of changes out for reviewers.

The old version worked because all of the supported logical type annotations had an equivalent `ConvertedType` (which is what `OriginalType` is called in Parquet format and the [logical type docs](https://github.com/apache/parquet-format/blob/master/LogicalTypes.md#numeric-types)).